---
title: "**4. Results**"
author: "Belinda Cornes"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,warning=FALSE)

library("kableExtra")
library("knitr")
options(stringsAsFactors = FALSE)
library(data.table) 
library(tidyr)     
library(mclust)     
library(rhdf5)      
library(optparse)
library(dplyr)
library(cluster)
library(readxl)
library(psych)
library(tibble)
library(ggplot2)
library(reshape2)
library(qtl2)
#library(qtl)
library(abind)

gmap <- read_csv("/Users/corneb/Documents/MyJax/CS/Projects/Pinkney/scripts/gmap.csv")
pmap <- read_csv("/Users/corneb/Documents/MyJax/CS/Projects/Pinkney/scripts/pmap.csv")
phenos <- read_csv("/Users/corneb/Documents/MyJax/CS/Projects/Pinkney/scripts/pheno_cleaned.csv")
genos <- read_csv("/Users/corneb/Documents/MyJax/CS/Projects/Pinkney/scripts/geno.csv")
covars <- read_csv("/Users/corneb/Documents/MyJax/CS/Projects/Pinkney/scripts/pheno_covar.csv")

cross <- qtl2::read_cross2("/Users/corneb/Documents/MyJax/CS/Projects/Pinkney/scripts/gm_cleaned.json")
select_phenos <- colnames(cross$pheno)
df_phenos <- cross$pheno

phenos$id <- rownames(phenos)
covars$id <- rownames(covars)
phenos.covars <- merge(phenos[c("id","R_AVG","L_AVG","Both_AVG","deg_avg_r", "deg_avg_l" ,"deg_avg_b")],covars[c("id","Sex")], by=c("id"), all=T)
##reformat
phenos.covars.lg <- phenos.covars %>% gather(variable, value, -c("id","Sex"))
phenos.covars.1 <- subset(phenos.covars.lg, variable %in% c("R_AVG","L_AVG","Both_AVG"))
phenos.covars.2 <- subset(phenos.covars.lg, variable %in% c("deg_avg_r", "deg_avg_l" ,"deg_avg_b"))

```

### **4.1. Sex effect**
A two-sample t-test was performed to assess the effect of sex among phenotypes. The test measures whether the average (expected) phenotypic value differs significantly between female and male groups. 

H0:  The average of each phenotype for female and male may be assumed to be equal.

```{r Sex effect, fig.height = 6, fig.width = 9.5, fig.align = "center"}

####t_test
df_phenos1 = df_phenos

is_female <- data.frame(cross$is_female)
df_phenos1= cbind(df_phenos1, is_female)


welch.t.test <- list()
for (phe in colnames(df_phenos[,c("R_AVG","L_AVG","Both_AVG","deg_avg_r", "deg_avg_l" ,"deg_avg_b")])){

  test <- t.test(df_phenos1[df_phenos1$cross.is_female == TRUE, phe ], df_phenos1[df_phenos1$cross.is_female == FALSE, phe ])
  test.df = data.frame(phenotype=phe,
                       t.statistic=test$statistic,
                       df=test$parameter,
                       p.value= test$p.value,
                       conf.int.95 = paste0(test$conf.int[1],", ",test$conf.int[2]),
                       female_mean = data.frame(test$estimate[1])[1,1],
                       male_mean = data.frame(test$estimate[2])[1,1])
  welch.t.test[[phe]] <- test.df

}

all.t.tests <- do.call(rbind, welch.t.test)

rownames(all.t.tests) <- NULL

all.t.tests[c(1,4,6,7)] %>%
  kable(escape = F,align = c("lccc")) %>%
  kable_styling("striped", full_width = T) %>%
  column_spec(1, bold=TRUE)


```

1. We observe a large p-value (p-value > 0.05) for all phenotypes (after the removal of outliers), so we cannot reject the null hypothesis of identical average scores. 
2. No p-value was smaller than 0.05, so the null hypothesis of equal averages was not rejected for any trait. 
3. Therefore the sex covariate was not added to the analysis


### **4.2. Genome-wide scan**
For each of the phenotype analyzed, 1000 permutations were used for each model to obtain genome-wide LOD significance threshold for p < 0.01, p < 0.05, p < 0.10,  respectively. 

The table shows the estimated significance thresholds from permutation test. 

```{r permutation, fig.height = 6, fig.width = 9.5, fig.align = "center"}

map <- insert_pseudomarkers(cross$gmap, step=1)
pmap_interp = interp_map(map, cross$gmap, cross$pmap)

gmapdf <- as.data.frame(do.call(abind, map))
gmapdf$index <- 1:nrow(gmapdf)
gmapdf$marker <- rownames(gmapdf)
pmapdf <- as.data.frame(do.call(abind, pmap_interp))
pmapdf$marker <- rownames(pmapdf)
mapdf <- merge(gmapdf,pmapdf, by=c("marker"), all=T)
mapdf$chr = sapply(strsplit(mapdf$marker,'_'),function(x) x[1])
mapdf$chr = sub("^c","", mapdf$chr)
mapdf$chr = sapply(strsplit(mapdf$chr,'.loc'),function(x) x[1])
mapdf <- mapdf[order(mapdf$index),]
names(mapdf) <- c('marker','gmapdf','index','pmapdf','chr')

pr <- calc_genoprob(cross, map, error_prob=0.002)
apr <- genoprob_to_alleleprob(pr)

kinship <- calc_kinship(pr)
Xcovar <- get_x_covar(cross)
addcovar = model.matrix(~Sex, data = covars)[,-1]

#operm <- scan1perm(pr, cross$pheno[,c("R_AVG","L_AVG","Both_AVG","deg_avg_r", "deg_avg_l" ,"deg_avg_b")], Xcovar=Xcovar, n_perm=2000)
#operm <- scan1perm(pr, cross$pheno[,c("R_AVG","L_AVG","Both_AVG","deg_avg_r", "deg_avg_l" ,"deg_avg_b")], addcovar = addcovar, n_perm=2000)
operm <- scan1perm(pr, cross$pheno[,c("R_AVG","L_AVG","Both_AVG","deg_avg_r", "deg_avg_l" ,"deg_avg_b")], n_perm=2000)

summary_table<-data.frame(unclass(summary(operm, alpha=c(0.01,  0.05, 0.1))))

summary_table$significance.level <- rownames(summary_table)

rownames(summary_table) <- NULL

summary_table[c(7,1:6)] %>%
  kable(escape = F,align = c("cccccccc")) %>%
  kable_styling("striped", full_width = T) %>%
  column_spec(1, bold=TRUE)

```

The figures below show QTL maps for each phenotype, the red solid lines correspond to genome-wide significance thresholds of p < 0.01, p < 0.05 and p < 0.1, respectively. 

```{r Genome-wide scan, fig.height = 6, fig.width = 9.5, fig.align = "center"}

#out <- scan1(pr, cross$pheno[,c("R_AVG","L_AVG","Both_AVG","deg_avg_r", "deg_avg_l" ,"deg_avg_b")], Xcovar=Xcovar)
#out <- scan1(pr, cross$pheno[,c("R_AVG","L_AVG","Both_AVG","deg_avg_r", "deg_avg_l" ,"deg_avg_b")], addcovar = addcovar)
out <- scan1(pr, cross$pheno[,c("R_AVG","L_AVG","Both_AVG","deg_avg_r", "deg_avg_l" ,"deg_avg_b")])

summary_table<-data.frame(unclass(summary(operm, alpha=c(0.01,  0.05, 0.1))))

plot_lod<-function(out,map){
  for (i in 1:dim(out)[2]){
    #png(filename=paste0("/Users/chenm/Documents/qtl/Jai/",colnames(out)[i],  "_lod.png"))
    
    #par(mar=c(5.1, 6.1, 1.1, 1.1))
    ymx <- maxlod(out[,2]) # overall maximum LOD score
    plot(out, map, lodcolumn=i, col="slateblue", ylim=c(0, 10))
    #legend("topright", lwd=2, colnames(out)[i], bg="gray90")
    title(main = colnames(out)[i])
    for (j in 1: dim(summary_table)[1]){
      abline(h=summary_table[j, i],col="red")
      text(x=400, y =summary_table[j, i]+0.12, labels = paste("p=", row.names(summary_table)[j]))
    }
    #dev.off()
  }
}

plot_lod(out,map)

```

We observed QTLs for all phenotypes.

### **4.3. LOD peaks**

The table below shows QTL peaks associated with the phenotype. We use the 95% threshold from the permutations to find peaks. 

```{r LOD peaks, fig.height = 6, fig.width = 9.5, fig.align = "center"}

peaks<-find_peaks(out, map, threshold=3.3, drop=1.5)

rownames(peaks) <- NULL
peaks[] %>%
  kable(escape = F,align = c("ccccccc")) %>%
  kable_styling("striped", full_width = T) %>%
  column_spec(1, bold=TRUE)

peaks_mbl <- list()
#corresponding info in Mb
for(i in 1:nrow(peaks)){
  lodindex <- peaks$lodindex[i]
  lodcolumn <- peaks$lodcolumn[i]
  chr <- peaks$chr[i]
  lod <- peaks$lod[i]
  pos <- mapdf$pmapdf[which(mapdf$gmapdf == peaks$pos[i] & mapdf$chr == peaks$chr[i])]
  ci_lo <- mapdf$pmapdf[which(mapdf$gmapdf == peaks$ci_lo[i] & mapdf$chr == peaks$chr[i])]
  ci_hi <- mapdf$pmapdf[which(mapdf$gmapdf == peaks$ci_hi[i] & mapdf$chr == peaks$chr[i])]
  peaks_mb=cbind(lodindex, lodcolumn, chr, pos, lod, ci_lo, ci_hi)
  peaks_mbl[[i]] <- peaks_mb
}
peaks_mba <- do.call(rbind, peaks_mbl)
peaks_mba <- as.data.frame(peaks_mba)
peaks_mba[,c("chr", "pos", "lod", "ci_lo", "ci_hi")] <- sapply(peaks_mba[,c("chr", "pos", "lod", "ci_lo", "ci_hi")], as.numeric)

rownames(peaks_mba) <- NULL
peaks_mba[] %>%
  kable(escape = F,align = c("ccccccc")) %>%
  kable_styling("striped", full_width = T) %>%
  column_spec(1, bold=TRUE)

```
### **4.4. QTL effects**

For each peak LOD location in 4.3, we give the following information:

  1. raw phenotype values for each genotype group
  2. qtl effects for each genotype group
  3. additive and dominance effects
  4. list of genes

```{r QTL effects, results="asis"}

query_variants <- create_variant_query_func("/Users/corneb/Documents/MyJax/CS/Projects/QTL2_support.files/cc_variants.sqlite")
query_genes <- create_gene_query_func("/Users/corneb/Documents/MyJax/CS/Projects/QTL2_support.files/mouse_genes_mgi.sqlite")

for (i in 1:nrow(peaks)){
#for (i in 1:1){
  #Plot 1
  g <- maxmarg(pr, map, chr=peaks$chr[i], pos=peaks$pos[i], return_char=TRUE)
  #png(filename=paste0("/Users/chenm/Documents/qtl/Jai/","qtl_effect_", i, ".png"))
  #par(mar=c(4.1, 4.1, 1.5, 0.6))
  plot_pxg(g, cross$pheno[,peaks$lodcolumn[i]], ylab=peaks$lodcolumn[i], sort=FALSE)
  title(main = paste("chr", chr=peaks$chr[i], "pos", peaks$pos[i], "(",peaks$lodcolumn[i],")"), line=0.2)
  #dev.off()

  chr = peaks$chr[i]

# Plot 2
  #coeff <- scan1coef(pr[,chr], cross$pheno[,peaks$lodcolumn[i]], addcovar = addcovar)
  #coeff <- scan1coef(pr[,chr], cross$pheno[,peaks$lodcolumn[i]], Xcovar=Xcovar)
  coeff <- scan1coef(pr[,chr], cross$pheno[,peaks$lodcolumn[i]])
  plot_coef(coeff, 
       map[chr], columns=1:3,
       bgcolor="gray95", legend="bottomleft", 
       main = paste("chr", chr=peaks$chr[i], "pos", peaks$pos[i], "(",peaks$lodcolumn[i],")")
       )
  #last_coef <- unclass(coeff)[nrow(coeff),1:3]
  #for(t in seq(along=last_coef))
  #axis(side=4, at=last_coef[t], names(last_coef)[t], tick=FALSE)


 # Plot 3
  c2effB <- scan1coef(pr[,chr], cross$pheno[,peaks$lodcolumn[i]], contrasts=cbind(mu=c(1,1,1), a=c(-1, 0, 1), d=c(0, 1, 0)))
  #c2effB <- scan1coef(pr[,chr], cross$pheno[,peaks$lodcolumn[i]], addcovar = addcovar, contrasts=cbind(mu=c(1,1,1), a=c(-1, 0, 1), d=c(0, 1, 0)))
  #c2effB <- scan1coef(pr[,chr], cross$pheno[,peaks$lodcolumn[i]],Xcovar=Xcovar, contrasts=cbind(mu=c(1,1,1), a=c(-1, 0, 1), d=c(0, 1, 0)))
  plot(c2effB, map[chr], columns=2:3,
       bgcolor="gray95", legend="bottomleft", 
       main = paste("chr", chr=peaks$chr[i], "pos", peaks$pos[i], "(",peaks$lodcolumn[i],")")
       )
  #last_coef <- unclass(c2effB)[nrow(c2effB),2:3] # last two coefficients
  #for(t in seq(along=last_coef))
  #  axis(side=4, at=last_coef[t], names(last_coef)[t], tick=FALSE)


  #Table 1
  chr = peaks_mba$chr[i]
  start=peaks_mba$ci_lo[i]
  end=peaks_mba$ci_hi[i]

  genesgss = query_genes(chr, start, end)

  rownames(genesgss) <- NULL
  genesgss$strand_old = genesgss$strand
  genesgss$strand[genesgss$strand=="+"] <- "positive"
  genesgss$strand[genesgss$strand=="-"] <- "negative"

  #genesgss <- 
  #table <- 
  #genesgss[,c("chr","type","start","stop","strand","ID","Name","Dbxref","gene_id","mgi_type","description")] %>%
  #kable(escape = F,align = c("ccccccccccc")) %>%
  #kable_styling("striped", full_width = T) #%>% 
  #cat #%>%
  #column_spec(1, bold=TRUE)
#
  #print(kable(genesgss[,c("chr","type","start","stop","strand","ID","Name","Dbxref","gene_id","mgi_type","description")], escape = F,align = c("ccccccccccc")))

  print(kable(genesgss[,c("chr","type","start","stop","strand","ID","Name","Dbxref","gene_id","mgi_type","description")], "html") %>% kable_styling("striped", full_width = T))

  #table
  

}

```